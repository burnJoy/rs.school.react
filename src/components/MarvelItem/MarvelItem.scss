.marvel-item {
  width: 200px;
  border: 1px solid #fff;
  margin: 0 20px 40px;
  border-radius: 8px;
  overflow: hidden;

  &__img-wrap {
    height: 200px;
  }
  &__img {
    max-width: 100%;
    object-fit: cover;
    height: 200px;
  }

  &__info {
    padding: 8px;
    margin-top: 8px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
  }

  &__name {
  }

  &__link {
    margin-top: 8px;
    margin-left: auto;
  }
}
