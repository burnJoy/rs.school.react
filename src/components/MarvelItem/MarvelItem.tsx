import React from 'react';
import { Hero } from '../../types';
import './MarvelItem.scss';

type Props = {
  hero: Hero;
};

export default class MarvelItem extends React.Component<Props> {
  render() {
    const { hero } = this.props;
    return (
      <li className="marvel-item">
        <div className="marvel-item__img-wrap">
          <img
            className="marvel-item__img"
            src={`${hero.thumbnail.path}.${hero.thumbnail.extension}`}
            alt={hero.name}
          />
        </div>
        <div className="marvel-item__info">
          <div className="marvel-item__name">{hero.name}</div>
          <a className="marvel-item__link" href={hero.urls[0].url} target="_blank" rel="noreferrer">
            link
          </a>
        </div>
      </li>
    );
  }
}
