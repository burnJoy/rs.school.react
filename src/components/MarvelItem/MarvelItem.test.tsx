import React from 'react';
import { render, screen } from '@testing-library/react';
import MarvelItem from './MarvelItem';
import type { Hero } from '../../types';

describe('MarvelItem', () => {
  const hero: Hero = {
    id: '123',
    name: 'Spider-Man',
    thumbnail: {
      path: 'https://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
      extension: 'jpg',
    },
    urls: [
      {
        url: 'http://marvel.com/comics/characters/1009610/spider-man?utm_campaign=apiRef&utm_source=0d3db3a575ecd5ce4491ee1295b966c5',
      },
    ],
  };

  it('renders the hero name', () => {
    render(<MarvelItem hero={hero} />);
    expect(screen.getByText('Spider-Man')).toBeInTheDocument();
  });

  it('renders the hero image', () => {
    render(<MarvelItem hero={hero} />);
    expect(screen.getByAltText('Spider-Man')).toBeInTheDocument();
  });

  it('renders the hero link', () => {
    render(<MarvelItem hero={hero} />);
    expect(screen.getByText('link')).toHaveAttribute('href', hero.urls[0].url);
  });

  it('matches the snapshot', () => {
    const { container } = render(<MarvelItem hero={hero} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
