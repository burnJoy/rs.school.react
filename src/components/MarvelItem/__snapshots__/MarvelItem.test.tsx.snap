// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarvelItem > matches the snapshot 1`] = `
<li
  class="marvel-item"
>
  <div
    class="marvel-item__img-wrap"
  >
    <img
      alt="Spider-Man"
      class="marvel-item__img"
      src="https://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
    />
  </div>
  <div
    class="marvel-item__info"
  >
    <div
      class="marvel-item__name"
    >
      Spider-Man
    </div>
    <a
      class="marvel-item__link"
      href="http://marvel.com/comics/characters/1009610/spider-man?utm_campaign=apiRef&utm_source=0d3db3a575ecd5ce4491ee1295b966c5"
      rel="noreferrer"
      target="_blank"
    >
      link
    </a>
  </div>
</li>
`;
